### PR TITLE
mastodon: read max toot size from v2 API

### PR DESCRIPTION
**Pull request by jpcaruana**
relates to https://github.com/snarfed/bridgy/issues/1533

### CODE REVIEW

File: models.py [hunk 1]
```diff
@@ -247,11 +247,15 @@ def __getattr__(self, name):
...
+        j = json_loads(inst)
+        truncate_text_length = j.get("configuration", {}).get('statuses', {}).get('max_characters', None) or j.get('max_toot_chars', None)
+      else:
+        truncate_text_length = None
       kwargs = {
         'user_id': json_loads(auth_entity.user_json).get('id'),
         # https://docs-develop.pleroma.social/backend/API/differences_in_mastoapi_responses/#instance
-        'truncate_text_length':
-          json_loads(inst).get('max_toot_chars') if inst else None,
+        'truncate_text_length': truncate_text_length,
```
Comment (jpcaruana) [inline models.py:258]: I can keep a one liner here if you prefer, but I found it difficult to read

```python
'truncate_text_length': (json_loads(inst).get("configuration", {}).get('statuses', {}).get('max_characters', None) or json_loads(inst).get('max_toot_chars')) if inst else None
```

File: models.py [hunk 2]
```diff
@@ -247,11 +247,15 @@ def __getattr__(self, name):
     elif self.key.kind() == 'Mastodon':
       args = (auth_entity.instance(),) + args
       inst = auth_entity.app.get().instance_info
+      if inst:
+        j = json_loads(inst)
+        truncate_text_length = j.get("configuration", {}).get('statuses', {}).get('max_characters', None) or j.get('max_toot_chars', None)
```
Comment (snarfed) [inline models.py:252]: minor style nit for the future, you can drop the two `, None`s here

**Review by snarfed — APPROVED** [review]
This is great! Thanks especially both for keeping the backward compatibility, and for the new test.

**Comment by snarfed** [comment]
Thank you for the contribution! Merging and deploying now...

**Comment by snarfed** [comment]
Oh also I checked a few recent `MastodonApp` objects in the datastore, and they all have the v2 `configuration.statuses.max_characters` field, as expected.

File: models.py (hunk 2 above)
Comment (jpcaruana) [reply models.py:252]: sorry, with timezones delays, I could not remove the `None` here in time (as that's the default from `dict.get`, you're right)

Comment (snarfed) [reply models.py:252]: No worries! I merged it immediately after commenting, I didn't expect you to. Thanks again!

**Pull request merged**